RNA: Add Jetpack footer

kangzj opened the pull request:
#### Changes proposed in this Pull Request:
Added two components,
1. A8C SVG Title: 'An Automattic Airline'
2. Jetpack Admin footer, uses the SVG title above and a Jetpack branding
3. Added `plugin:@wordpress/eslint-plugin/i18n` to check i18n text domain.

#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Go to `projects/js-packages/components` and run `pnpm run test`
- Ensure unit tests pass


- Checkout #20252  and follow test instructions
- Ensure components in footer are rendered properly

- Ensure eslint fails if i18n text domain is set to strings other than `jetpack`

![image](https://user-images.githubusercontent.com/1425433/124697469-e485c000-df3a-11eb-8616-43961361b55d.png)

kangzj committed:
[not verified] a8c title svg component

kangzj committed:
[not verified] set svg size smaller

kangzj committed:
[not verified] added jetpack admin footer

kangzj committed:
[not verified] refactoring

kangzj committed:
[not verified] changelog

kangzj committed:
[not verified] fixed typo

kangzj committed:
[not verified] init jetpack-components 0.0.2

jeherve commented on projects/js-packages/components/components/a8c-svg-title/index.jsx:
```diff
@@ -0,0 +1,40 @@
+ * @returns {React.Component} A8cSvgTitle component.
+ */
+export default function A8cSvgTitle( {
+	title = 'An Automattic Airline',
```
Could you make this translatable with `@wordpress/i18n` and a Jetpack textdomain?

kangzj replied:
addressed in 05bed62

jeherve commented on projects/js-packages/components/components/a8c-svg-title/index.jsx:
```diff
@@ -0,0 +1,40 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
```
Would it make sense to use the WordPress abstraction from `@wordpress/element` instead of React itself here?

kangzj replied:
Another component `JetpackLogo` depends directly on `react` as well, so not to make this PR more complicated. I'll discussion with @sergeymitr and see how we'll do it. Thanks

jeherve commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+ *
+ * @param {object} props - Component properties.
+ * @param {object} props.aboutPageUrl - Link for 'An Automattic Airline'.
+ * @param {object} props.moduelName - Name of the moduel, e.g. 'Jetpack Search'.
```
Maybe worth fixing that typo (moduel -> module) here and below?

kangzj replied:
fixed in 38f33d9

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/style.scss:
```diff
@@ -0,0 +1,31 @@
+	}
+}
+.jp-dashboard-footer__logo:before {
+	font-family: jetpack !important;
```
Is there a way to avoid using `!important` here?

kangzj replied:
`!important` is not actually essential here. I just wanted to make sure, the font is overridden properly. Now removed and it works fine.

jsnmoon commented on projects/js-packages/components/components/a8c-svg-title/README.md:
```diff
@@ -0,0 +1,16 @@
+# A8C SVG Title
```
Could we use a more descriptive name for the component, like `automattic-byline-logo`? I'm not sure having `svg` in the component name is necessary since that's self-evident.

kangzj replied:
renamed in 25d1d9583

jsnmoon commented on projects/js-packages/components/components/a8c-svg-title/README.md:
```diff
@@ -0,0 +1,16 @@
+
+#### Props
+
+- `className`: String - (default: `jp-a8c-svg-title`) the class name set on the SVG element.
```
Instead of overriding `.jp-a8c-svg-title`, might I recommend that we use the `classNames` module to allow developers to add to the existing class name like we do for the footer component below? Users of component libraries typically expect to be able to append class names onto components without losing the built-in styling.

kangzj replied:
addressed in 4b25a83

jsnmoon reviewed (commented):
Looks like this needs a rebase/master merge; diff currently includes changes from #20288.

kangzj committed:
[not verified] rename a8c-svg-title to automattic-byline-logo

kangzj committed:
[not verified] use classnames for automattic logo component

kangzj committed:
[not verified] fixed typo

kangzj committed:
[not verified] i18n logo title

kangzj committed:
[not verified] removed important

kangzj commented:
> Looks like this needs a rebase/master merge; diff currently includes changes from #20288.

Thank you @jsnmoon and @jeherve for the review 👍 . Rebased and addressed feedback.

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+ */
+export default function JetpackFooter( {
+	aboutPageUrl,
+	moduleName = 'Jetpack',
```
Does this also need to be translated with `__`? 🙂

kangzj replied:
Good point. I did think about it. I thought it's our brand and since wouldn't need to be translated. But a brief search across our code base, found a couple of cases translating `Jetpack`. I guess for most languages it is kept just as is. So added translation in 998e17337

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+ *
+ * @param {object} props - Component properties.
+ * @param {object} props.aboutPageUrl - Link for 'An Automattic Airline'.
+ * @param {object} props.moduleName - Name of the moduel, e.g. 'Jetpack Search'.
```
```suggestion
 * @param {object} props.moduleName - Name of the module, e.g. 'Jetpack Search'.
```

jsnmoon replied:
Idea: What do you think about calling this property something like `title`?

jsnmoon reviewed (commented):
Found places for small tweaks here and there. Nonetheless, this looks ready for crew review!

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+ * JetpackFooter component definition.
+ *
+ * @param {object} props - Component properties.
+ * @param {object} props.aboutPageUrl - Link for 'An Automattic Airline'.
```
Idea: What do you think about calling this property `logoHref`?

kangzj replied:
hmm the left-hand side is Jetpack logo and the right is the A8C one, so`logoHref` might be a bit ambiguous. Will change it to `a8cLogHerf` if it's okay. 😄

kangzj committed:
[not verified] added translation

kangzj committed:
[not verified] renamed aboutPageUrl to a8cLogoHref

kangzj committed:
[not verified] fixed typo

kangzj commented:
@jwebbdev is working the standalone plugin which doesn't load the `jetpack` font and requested to change the jetpack logo to SVG. Since this PR is already half way down the road, I'll change it with PR.

jeherve commented on projects/js-packages/components/components/automattic-byline-logo/index.jsx:
```diff
@@ -0,0 +1,42 @@
+ * @returns {React.Component} AutomatticBylineLogo component.
+ */
+export default function AutomatticBylineLogo( {
+	title = __( 'An Automattic Airline' ),
```
You'll want to add the `jetpack` textdomain here. 

```suggestion
	title = __( 'An Automattic Airline', 'jetpack' ),
```

Maybe that's a good opportunity to add the `plugin:@wordpress/eslint-plugin/i18n` plugin to our eslint config for Jetpack components?
https://github.com/Automattic/jetpack/blob/3128bbedf42abc396d0070cbe1e2e08f6d982461/projects/js-packages/components/.eslintrc.cjs#L4


*******

Somewhat related, as we start using those packages in other plugins, we'll have to handle updating those textdomains as we release. Tracking this in #20320.

kangzj replied:
>You'll want to add the jetpack textdomain here.

Ops thanks for catching it.

>Maybe that's a good opportunity to add the plugin:@wordpress/eslint-plugin/i18n plugin to our eslint config for Jetpack components?

Sure. will do.

kangzj replied:
Hi @jeherve added `plugin:@wordpress/eslint-plugin/i18n` in cf9f213. Could you please double check if it works as expected? Thanks 😄

jeherve replied:
It does, thank you!

<img width="1019" alt="Screenshot 2021-07-12 at 10 36 51" src="https://user-images.githubusercontent.com/426388/125256850-21680300-e2fd-11eb-912e-02416d67cf72.png">

kangzj committed:
[not verified] Update projects/js-packages/components/components/automattic-byline-logo/index.jsx

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/README.md:
````diff
@@ -0,0 +1,20 @@
+```js
+<JetpackFooter
+	moduleName="Jetpack Search"
+	a8cLogoHerf="https://www.jetpack.com"
````
```suggestion
	a8cLogoHref="https://www.jetpack.com"
```

Herf! 🤣 

(TIL `href` means "Hypertext Reference")

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/README.md:
```diff
@@ -0,0 +1,20 @@
+#### Props
+
+- `className`: String - (default: `jp-dashboard-footer`) the additional class name set on the element.
+- `a8cLogoHerf`: String - (default: `https://www.jetpack.com`) link to be added on 'An Automattic Airline'.
```
```suggestion
- `a8cLogoHref`: String - (default: `https://www.jetpack.com`) link to be added on 'An Automattic Airline'.
```

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+ * @returns {React.Component} JetpackFooter component.
+ */
+export default function JetpackFooter( {
+	a8cLogoHerf,
```
```suggestion
	a8cLogoHref,
```

jsnmoon commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+				<span>{ moduleName }</span>
+			</div>
+			<div className="jp-dashboard-footer__footer-right">
+				<a href={ a8cLogoHerf }>
```
```suggestion
				<a href={ a8cLogoHref }>
```

They call me the Herf hunter 😄

jsnmoon reviewed (dismissed):
Found some typos!

kangzj committed:
[not verified] Update projects/js-packages/components/components/jetpack-footer/README.md

Co-authored-by: Jason Moon <4044428+jsnmoon@users.noreply.github.com>

kangzj committed:
[not verified] Update projects/js-packages/components/components/jetpack-footer/index.jsx

Co-authored-by: Jason Moon <4044428+jsnmoon@users.noreply.github.com>

kangzj committed:
[not verified] Update projects/js-packages/components/components/jetpack-footer/index.jsx

Co-authored-by: Jason Moon <4044428+jsnmoon@users.noreply.github.com>

kangzj committed:
[not verified] Update projects/js-packages/components/components/jetpack-footer/README.md

Co-authored-by: Jason Moon <4044428+jsnmoon@users.noreply.github.com>

kangzj committed:
[not verified] fixed typo

kangzj committed:
[not verified] add i18n check to eslint

jeherve commented on projects/js-packages/components/components/jetpack-footer/index.jsx:
```diff
@@ -0,0 +1,43 @@
+			</div>
+			<div className="jp-dashboard-footer__footer-right">
+				<a href={ a8cLogoHref }>
+					<AutomatticBylineLogo title={ __( 'An Automattic Airline', 'jetpack' ) } />
```
Since our title is the same as the default, do we need to include it here?

```suggestion
					<AutomatticBylineLogo />
```

jeherve reviewed (commented):
This looks good to me. I have one question here.

kangzj committed:
[not verified] Update projects/js-packages/components/components/jetpack-footer/index.jsx

Co-authored-by: Jeremy Herve <jeremy@jeremy.hu>

kangzj commented:
> This looks good to me. I have one question here.

Good point. Committed. And also update https://github.com/Automattic/jetpack/pull/20252 for testing. Thanks 👍

jeherve reviewed (approved):
This looks good to me, although I can't seem to see the footer at all when using the Beta plugin; I only see it on my local installation. I think it may be that the components do not get built when using the Beta plugin?

In any case, that's probably not something to address in this PR. Logged this in #20330.

kangzj commented:
🚢 r228524-wpcom

sergeymitr commented:
Hi @kangzj.
Good job on building the component.
We're about to build a similar one for header, and there'll likely be other layout components later on.

So I'm thinking that instead of putting them all into `jetpack-components`, we should group them all in the new package `jetpack-layout`.

Do you see any potential undesired consequences of doing that?
I didn't manage to find any consumers of the footer component, are there any?

kangzj commented:
> So I'm thinking that instead of putting them all into `jetpack-components`, we should group them all in the new package `jetpack-layout`.
> 
> Do you see any potential undesired consequences of doing that?
> I didn't manage to find any consumers of the footer component, are there any?

Hi @sergeymitr sure thing, I think it will be clearer. Tbh we are super happy just to consume them :-)

- The search admin page is using it https://github.com/Automattic/jetpack/pull/20252
- I think @jwebbdev is going to use it for the Backup plugin

kangzj commented:
Hi @sergeymitr

>Would it make sense to use the WordPress abstraction from @wordpress/element instead of React itself here? --@jeherve

Just while you are here, are we going to import @wordpress/element instead of React?
